lib/npg_pipeline/daemons/base.pm

```diff
@@ -33,37 +29,61 @@ Readonly::Scalar my $NO_LIMS_LINK => -1;
 class_has 'pipeline_script_name' => (
   isa        => q{Str},
   is         => q{ro},
+  metaclass  => 'NoGetopt',
   lazy_build => 1,
 );
-sub _build_pipeline_script_name {
-  return $PIPELINE_SCRIPT;
-}
+
+has 'dry_run' => (
+  isa        => q{Bool},
+  is         => q{ro},
+  required   => 0,
+  default    => 0,
+  documentation => 'Dry run mode flag, false by default',
+);
+
+has 'logger' => (
+  isa        => q{Log::Log4perl::Logger},
+  is         => q{ro},
+  metaclass  => 'NoGetopt',
+  default    => sub { Log::Log4perl->get_logger() },
+);
+
+around 'log' => sub {
+   my $orig = shift;
+   my $self = shift;
+   $self->logger->info('"log" method is depricated');
```

Comment:
> + $self->logger->info('"log" method is depricated');
>
> deprecated
>
Thank you, fixed.




-- 
 The Wellcome Trust Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.